pyquil/api/client/compiler/_compiler_client.py

```diff
@@ -0,0 +1,237 @@
+##############################################################################
+# Copyright 2016-2021 Rigetti Computing
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+##############################################################################
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import Iterator, Optional, cast, List
+
+import rpcq
+from rpcq.messages import TargetDevice as TargetQuantumProcessor
+
+
```

Comment:
@notmgsk Could use a deeper look from you at the docstrings in this file, to check for accuracy